pkg/api/authn.go

```diff
@@ -198,84 +349,188 @@ func basicAuthHandler(ctlr *Controller) mux.MiddlewareFunc {
 				return
 			}
 
-			// we want to bypass auth for mgmt route
-			isMgmtRequested := request.RequestURI == constants.FullMgmtPrefix
+			//nolint: contextcheck
+			passed, cloneResp, cloneReq, err := am.sessionAuthn(ctlr, response, request)
+			if err != nil {
+				response.WriteHeader(http.StatusInternalServerError)
+
+				return
+			}
 
-			if request.Header.Get("Authorization") == "" {
-				if anonymousPolicyExists(ctlr.Config.HTTP.AccessControl) || isMgmtRequested {
-					// Process request
-					ctx := getReqContextWithAuthorization("", []string{}, request)
-					next.ServeHTTP(response, request.WithContext(ctx)) //nolint:contextcheck
+			if passed && cloneResp != nil && cloneReq != nil {
+				next.ServeHTTP(cloneResp, cloneReq)
 
-					return
-				}
+				return
 			}
 
-			username, passphrase, err := getUsernamePasswordBasicAuth(request)
+			//nolint: contextcheck
+			passed, cloneResp, cloneReq, err = am.basicAuthn(ctlr, response, request)
```

Comment:
why is basicAuth second to sessionAuth - what if user already has a previous session but is making an action with other identity ( new credentials)? Will it be needed to clear the session first ( i.e. log out) to proceed with this action ?

Reply:
yes, that's right, didn't think about it, will modify.